### PR TITLE
New main panel layout and date filter

### PR DESCRIPTION
**Pull request by doug24**
Issue #1 and Filters panel toggle, and wildcard available on main screen
Issue #221 Filter by date
Issue #251 ClearType on

Note that the File Options panel will toggle open and closed with the alt+F key.

It was suggested to put the Whole Word/Match Case options under the Filters expander, but I prefer to have them on the main panel.  So I created a new option to let the user choose the location they prefer:  change the location using the "Show search options" selection in the User Interface tab of the Options dialog.

### CODE REVIEW

**Comment by JVimes** [comment]
Super exciting! Will review this as soon as I can.

File: dnGREP.WPF/MVHelpers/BoolToVisibilityConverter.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Globalization;
+using System.Windows;
+using System.Windows.Data;
+
+namespace dnGREP.WPF
+{
+    [ValueConversion(typeof(bool), typeof(Visibility))]
+    public class BoolToVisibilityConverter : IValueConverter
```
Comment (JVimes) [inline dnGREP.WPF/MVHelpers/BoolToVisibilityConverter.cs]: WPF has a built-in [BooleanToVisibilityConverter](https://msdn.microsoft.com/en-us/library/system.windows.controls.booleantovisibilityconverter%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396) type you can instantiate right in a resource dictionary:

    <BooleanToVisibilityConverter x:Key="BooleanToVisibilityConverter" />

It uses `Collapsed`. Looks like that would eliminate the need for this file?

Comment (JVimes) [inline dnGREP.WPF/MVHelpers/BoolToVisibilityConverter.cs]: Oh, looks like the built-in one was replaced with this. There must be a difference (this one handles nullable bool, maybe?)

Comment (doug24) [inline dnGREP.WPF/MVHelpers/BoolToVisibilityConverter.cs]: This new one handles both bool-to-visibility and inverse-bool-to-visibility in one converter - so the code can toggle between the two sets of check box locations using the same data binding flag.  I could have just added a new inverse-bool-to-visibility converter instead.

**Pull request merged**